src/DocBlock/TypeExpression.php

```diff
@@ -56,14 +56,14 @@ final class TypeExpression
             (?:
                 (?<array_shape>
                     (?<array_shape_name>(?i)(?:array|list|object)(?-i))
-                    (?<array_shape_start>\h*\{\h*)
+                    (?<array_shape_start>\h*\{[\h\v*]*)
```

Comment:
I think the main regex for types should not support `*`. If this approach is needed, keep the original regex as is and let's have a 2nd one with something like `str_replace('\h', '[\h\v*]', ...`.

Reply:
> I think the main regex for types should not support `*`. `*` is not part of phpdoc grammar.

How `*` is not part of phpdoc grammar?

Reply:
I mean part of the phpdoc type.

Reply:
Multiline PHPDoc types are, in fact, types "polluted" with whitespaces and `*`.

I wanted the simplest - in terms of changes - solution, feel free to propose an alternative with `str_replace('\h', '[\h\v*]', ...)`.

Reply:
I mean to have a static method returning the regex with the solution you propose now and cached. This way I would be happy as the regex will remain untouched and someone else can impl. the idea from https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8893#discussion_r2238017321 later.

Reply:
I understand what you want to have, but I simply don't agree with it. The issue was opened in 2021, so anyone had time to propose a solution, wouldn't you agree?

I have proposed one, so let's focus on whether it is correct, not if you like it.

Reply:
> I say this is wrong

And I say it is right, which is as much as my opinion as yours is your opinion.

You say a lot of "should", "must", etc. - I like to stick to the facts, and those are:
- no other proposition in over 3 years
- new feature is supported
- nothing gets broken

Reply:
for sake of solving issue from _2021_, i'm OK to merge this as-is.
if you have even better proposal how to fix this @mvorisek , please raise follow-up PR, unless it would make things over-complicated

Reply:
I understand you guys but I simply view things differently. I work mostly on databases where integrity is important and no compromises are allowed. This tool is a piece of art and I probably should hand over my work on this topic to the next hands. Enjoy the summer! 🏝️

Reply:
i see value in your original idea, we simply need to get pragmatism when it's possible, otherwise the issue will stay open another 4 years. "Perfect is the enemy of good" and "we released in perfect way, just 5 years delayed"